api/services_cluster.go

```diff
@@ -0,0 +1,100 @@
+package api
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+	"github.com/gorilla/mux"
+
+	"github.com/canonical/microcloud/microcloud/api/types"
+	"github.com/canonical/microcloud/microcloud/service"
+)
+
+// ServicesClusterCmd represents the /1.0/services/cluster/{name} API on MicroCloud.
+var ServicesClusterCmd = func(sh *service.Handler) rest.Endpoint {
+	return rest.Endpoint{
+		AllowedBeforeInit: true,
+		Name:              "services/cluster/{name}",
+		Path:              "services/cluster/{name}",
+
+		Delete: rest.EndpointAction{Handler: authHandler(sh, removeClusterMember), AllowUntrusted: true},
+	}
+}
+
+// removeClusterMember removes the given cluster member from all services that it exists in.
+func removeClusterMember(state *state.State, r *http.Request) response.Response {
+	force := r.URL.Query().Get("force") == "1"
+	name, err := url.PathUnescape(mux.Vars(r)["name"])
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	supportedServices := map[types.ServiceType]string{
+		types.MicroOVN:  MicroOVNDir,
+		types.MicroCeph: MicroCephDir,
+		types.LXD:       LXDDir,
+	}
+
+	existingServices := []types.ServiceType{types.MicroCloud}
+	for serviceType, stateDir := range supportedServices {
+		if service.Exists(serviceType, stateDir) {
+			existingServices = append(existingServices, serviceType)
+		}
+	}
+
+	addr, _, err := net.SplitHostPort(state.Address().URL.Host)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("State address %v is invalid: %w", state.Address().String(), err))
+	}
+
+	sh, err := service.NewHandler(state.Name(), addr, state.OS.StateDir, false, false, existingServices...)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Remove the node from services in the following order:
+	// 1. Remove from LXD first as it may have storage & networks that depend on the others for cleanup.
+	// 2. Remove from MicroCeph and MicroOVN next, concurrently.
+	// 3. Remove from MicroCloud last so that if there were any errors causing the other services to fail, MicroCloud will still know about the node.
+	var memberExists bool
+	err = sh.RunConcurrent(types.LXD, types.MicroCloud, func(s service.Service) error {
+		existingMembers, err := s.ClusterMembers(r.Context())
+		if err != nil && !api.StatusErrorCheck(err, http.StatusServiceUnavailable) {
+			return err
+		}
+
+		// If we got a 503 error back, that means the service is installed, but hasn't been set up yet, so there are no cluster members to remove.
+		if err != nil {
+			return nil
+		}
```

Comment:
Would this be a little bit easier to read with a nested `if`? Not a blocker though.

```go
if err != nil {
    if api.StatusErrorCheck(err, http.StatusServiceUnavailable) {
        return nil
    }

    return err
}
```

Reply:
I'm usually not a big fan of nesting in `if err != nil` as it's easy to overlook when you're not familiar with it.